lib/security/index.js

```diff
@@ -0,0 +1,110 @@
+const { signJWT } = require('./jwt')
+
+async function parseSecurity (oasSecuritySchemes, securityRequirements, securityOptions) {
+  const headers = {}
+  const queries = {}
+
+  // We only need to set these values if there are security schemes defined within the OAS
+  if (oasSecuritySchemes) {
+    // For each required security mechanism
+    for await (const key of Object.keys(securityRequirements)) {
+      const securityScheme = oasSecuritySchemes[key]
+
+      // Security scheme is not defined in the global scheme, throw error
+      if (!securityScheme) {
+        throw new Error(`Security scheme ${key} not defined in spec.`)
+      }
+
+      const { secret, jwt } = securityOptions
+
+      /**
+       * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#security-scheme-object
+       *
+       * REQUIRED. The type of the security scheme. Valid values are "apiKey", "http", "oauth2", "openIdConnect".
+       *
+       * Only apiKey and http options are implemented here
+       */
+      switch (securityScheme.type) {
+        // http type
+        case 'http':
+          if (!headers.authorization) {
+            headers.authorization = []
+          }
+          /**
+           * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#security-scheme-object
+           *
+           * REQUIRED. The name of the HTTP Authorization scheme to be used in the Authorization header as defined in RFC7235.
+           * The values used SHOULD be registered in the IANA Authentication Scheme registry.
+           *
+           * https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml
+           *
+           * Only bearer is implemented here
+           */
+          switch (securityScheme.scheme) {
+            case 'bearer':
+              /**
+               * A hint to the client to identify how the bearer token is formatted. Bearer tokens are usually
+               * generated by an authorization server, so this information is primarily for documentation purposes.
+               *
+               * Only the JWT mechanism is used here, otherwise, the token is just a plain value
+               */
+              switch (securityScheme.bearerFormat) {
+                case 'JWT': {
+                  const { exp } = jwt || {}
+                  const jwtOptions = {}
+                  if (exp) {
+                    jwtOptions.expiresIn = exp
+                  }
+
+                  headers.authorization.push(`Bearer ${await signJWT({}, secret, jwtOptions)}`)
```

Comment:
Right now we just sign on an blank object. Still need to discuss this. 

Sign blank object
Sign on body for non GET
Sign on query for GET

As the latter two sends more information than required.

Reply:
what if the `jwt` param in options indicates a `payload` property that can have either: `query`, `body`, or a custom object as needed?